### ISSUE TITLE
helm template's output is unstable (unordered) / namespace is not applied as first on helm template

### ISSUE THREAD

**Issue by mrueg** [helm/helm#4728]
Output of `helm version`: helm-2.10.0

Output of `kubectl version`: kubectl-1.12.0

Cloud Provider/Platform (AKS, GKE, Minikube etc.): minikube

assume the following chart:
```
minimal/namespace.yaml
minimal/deployment.yaml
minimal/service.yaml
minimal/serviceaccount.yaml
```

Run this command:
`for i in {1..100}; do diff -u <(helm template charts/minimal  2>&1 | grep "^kind") <(helm template charts/minimal 2>&1 | grep "^kind"); done`
You will see the Namespace being applied on arbitrary positions in the rendered template.
```
--- /proc/self/fd/11    2018-10-02 14:48:11.910223827 +0200
+++ /proc/self/fd/12    2018-10-02 14:48:11.910223827 +0200
@@ -1,4 +1,4 @@
-kind: Namespace
 kind: ServiceAccount
 kind: Service
 kind: Deployment
+kind: Namespace
--- /proc/self/fd/11    2018-10-02 14:48:11.978223823 +0200
+++ /proc/self/fd/12    2018-10-02 14:48:11.978223823 +0200
@@ -1,4 +1,4 @@
-kind: Namespace
 kind: ServiceAccount
 kind: Service
+kind: Namespace
 kind: Deployment
```

Rename namespace.yaml to 01-namespace.yaml:
```
minimal/01-namespace.yaml
minimal/deployment.yaml
minimal/service.yaml
minimal/serviceaccount.yaml
```

and the command above would not output anything.

As a user, I would like the `helm template` output apply in the same order as tiller would.

**Comment by mrueg** [comment]
I had helm-2.10.0 installed, it seems to be fixed with 2.11.0

**Comment by mrueg** [comment]
Unfortunately this doesn't appear to be fixed, output is still unstable

**Comment by filintod** [comment]
There is a logic issue on tiller.kind_sorter logic when Namespace is use (value 0 on the order).

The fix is:

	// if both are unknown
	if !aok && !bok {
		// and of different kind sort by kind alphabetically
		if a.Head.Kind != b.Head.Kind {
			return a.Head.Kind < b.Head.Kind
		}
		// and of same kind sort by name alphabetically
		return a.Name < b.Name
	}

	// if same kind sort alphanumeric
	if first == second {
		return a.Name < b.Name
	}

instead of

	// if same kind (including unknown) sub sort alphanumeric
	if first == second {
		// if both are unknown and of different kind sort by kind alphabetically
		if !aok && !bok && a.Head.Kind != b.Head.Kind {
			return a.Head.Kind < b.Head.Kind
		}
		// and of same kind sort by name alphabetically
		return a.Name < b.Name
	}

I'll put a pull request later

**Comment by bacongobbler** [comment]
fixed via #5186
